runtime/metricsview/ast.go

```diff
@@ -134,16 +141,30 @@ func NewAST(mv *runtimev1.MetricsViewSpec, sec *runtime.ResolvedSecurity, qry *Q
 	// Build dimensions to apply against the underlying SELECT.
 	// We cache these in the AST type because when resolving expressions and adding new JOINs, we need the ability to reference these.
 	ast.dimFields = make([]FieldNode, 0, len(ast.query.Dimensions))
+	timeFields := make([]FieldNode, 0, len(ast.query.Dimensions))
+	minGrain := TimeGrainYear
 	for _, qd := range ast.query.Dimensions {
 		dim, err := ast.resolveDimension(qd, true)
 		if err != nil {
 			return nil, fmt.Errorf("invalid dimension %q: %w", qd.Name, err)
 		}
 
+		tm, gn := ast.isTime(qd)
 		f := FieldNode{
 			Name:  dim.Name,
 			Label: dim.Label,
 			Expr:  ast.dialect.MetricsViewDimensionExpression(dim),
+			TimeProps: TimeProps{
+				Time:      tm,
+				TimeGrain: gn,
+				MinGrain:  gn,
+			},
+		}
+		if f.TimeProps.Time {
+			timeFields = append(timeFields, f)
+			if gn.isLess(minGrain) {
+				minGrain = gn
+			}
```

Comment:
In the interest of keeping the `NewAST` function simple, can we move the logic of computing the grain to `addTimeComparisonMeasure`? It's very efficient and unlikely to get called a lot, so should be fine in terms of performance.

Then I think the `FieldNode.TimeProps` can also be turned into just `Time bool`.